### DISCUSSION TITLE
Hide credentials in RTSP cameras and servers

### DISCUSSION THREAD

**Discussion by mgdelafuente** [bluenviron/mediamtx#3968] [Questions] [answered]
### Question

Hello everyone and sorry if this is a repeated question,

I am implementing mediamtx at my home to centralize connections to my cameras and to use rtsps. To do this I will only activate rtsps over tcp and using username and password with sha256 encryption.
The problem I see is that, when listing my cameras in the configuration file, I am forced to indicate the connection url with the format _"rtsp://user:password@ip:port/resource"._

Is there a way to hide the camera credentials in some way?

I can think of creating the configuration file just before running mediamtx and deleting the file as soon as it has been run. I could implement this using systemd through ExecStartPre, ExecStart and ExecStartPost.
Is there a better way to do this?

**Comment by mgdelafuente** [comment]
Also I see I can encrypt the configuration https://github.com/bluenviron/mediamtx?tab=readme-ov-file#encrypt-the-configuration

**Reply by aler9** [reply, marked as answer]
Exactly. Encrypt the configuration and specify a key when you launch the server.

**Reply by nh2** [reply]
Updated docs link: https://mediamtx.org/docs/usage/encrypt-the-configuration
